### DISCUSSION TITLE
HTTP status: 401 Failed to post comment. Status code: 401

### DISCUSSION THREAD

**Discussion by abhisek247767** [abhisek247767/Hello-World#98] [Q&A]
---
name: Auto Comment on PR Merge

on:
  pull_request:
    types: [closed]

jobs:
  comment:
    runs-on: ubuntu-latest
    if: github.event.pull_request.merged == true
    steps:
    - name: Checkout Repository
      uses: actions/checkout@v2
    
    - name: Debug Information
      run: |
        echo "Event name: ${{ github.event_name }}"
        echo "PR number: ${{ github.event.pull_request.number }}"
        echo "Merged: ${{ github.event.pull_request.merged }}"
    
    - name: Add Comment to PR
      env:
        GH_PAT: ${{ secrets.GH_PAT_COMMENT }}
      run: |
        COMMENT=$(cat <<EOF
        {
          "body": "🎉 Your pull request has been successfully merged! 🎉 Thank you for your valuable contribution to our project. Your efforts are greatly appreciated. Feel free to reach out if you have any more contributions or if there's anything else we can assist you with. Keep up the fantastic work! 🚀"
        }
        EOF
        )
        
        RESPONSE=$(curl -X POST \
          -H "Authorization: token $GH_PAT" \
          -H "Accept: application/vnd.github.v3+json" \
          "https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.pull_request.number }}/comments" \
          -d "$COMMENT" \
          -w "\n%{http_code}" \
          -s)
        HTTP_STATUS=$(echo "$RESPONSE" | tail -n1)
        BODY=$(echo "$RESPONSE" | sed '$d')
        
        echo "Response body: $BODY"
        echo "HTTP status: $HTTP_STATUS"
        
        if [ "$HTTP_STATUS" -ne 201 ]; then
          echo "Failed to post comment. Status code: $HTTP_STATUS"
          exit 1
        fi
        
        echo "Comment posted successfully!"
---
can someone ell why my job fail every time.. i tried with GitHub default secret token also still.

**Comment by DevrMichael** [comment]
The 401 means the token you’re using doesn’t have permission to post comments on PRs.

A few things to check:

1. Personal Access Token (PAT): Make sure `GH_PAT_COMMENT` has the repo and public_repo scopes (depending on your repo visibility).
2. Default GitHub Token: If you’re using `GITHUB_TOKEN`, it must have permissions: pull-requests: write set at the job or workflow level. Otherwise comment creation will fail.
3. Token usage: Ensure you use Authorization: Bearer ${{ secrets.GITHUB_TOKEN }} for the default token, not token.

Example fix:
```
permissions:
  pull-requests: write
```

Add that to the top of your workflow, and it should resolve the 401 error.
